File: 0428_week26/Programmers_카펫/KwonYi.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int area = brown + yellow;
        
        for(int c = 3; c <= area / 3; c++){
            if(area % c != 0){
                continue;
            }
            
            int r = area / c;
            
            if(2*r + 2*c - 4 == brown){
                answer[0] = r;
                answer[1] = c;
                break;
            }
        }
        return answer;
    }
}

File: 0714_week32/programmers_조이스틱/KwonYI.py
def solution(n, times):
    times.sort()
    answer = 100000 * 1000000000 + 1
    left, right = 1, n * times[-1]
    
    while left <= right :
        mid = (left + right) // 2
        
        if possible(times, n, mid) :
            answer = min(answer, mid)
            right = mid - 1
        else :
            left = mid + 1
    
    return answer

def possible(times, n, mid) :
    amount = sum(map(lambda x : mid // x, times))
    return amount >= n